flink-state-backends/flink-statebackend-changelog/src/main/java/org/apache/flink/state/changelog/ChangelogKeyedStateBackend.java

```diff
@@ -144,11 +143,9 @@
     private final HashMap<String, InternalKvState<K, ?, ?>> keyValueStatesByName;
 
     /**
-     * Unwrapped changelog states used for recovery (not wrapped into e.g. TTL).
-     *
-     * <p>WARN: cleared upon recovery completion.
+     * Unwrapped changelog states used for recovery (not wrapped into e.g. TTL, latency tracking).
      */
-    private final HashMap<String, ChangelogState> changelogStates;
+    private final Map<String, ChangelogState> changelogStates;
```

Comment:
Please either change all three `HashMap` to `Map` and put them into a separate hotfix commit
Or Leave it as it is.